code/datums/action.dm

```diff
@@ -500,6 +501,18 @@
 		return spell.can_cast(owner)
 	return 0
 
+/datum/action/spell_action/UpdateButtonIcon()
+	if(button && !(. = ..()))
```

Comment:
The `!(. = ..())` here with the assignment of `.` inside the if statement is weird, does this actually work properly? I would think it needs to be `(!..())` or something else with no assignment.

Reply:
In languages like C and C++ this is valid syntax. Same as in byond.
Written out it will act the same as:
```
if(!button)
  return
. = ..()
if(!.)
  stuf_here()
```
I prefer to write it like this due to the 3 lines it saves

Reply:
Well then, didn't know you could do that, carry on :smile: